packages/editor/src/editor/sidebar/fields/BadgeField.tsx

```diff
@@ -0,0 +1,49 @@
+import { BasicField, InputBadge, InputBadgeArea, splitNewLine, type BasicFieldProps } from '@axonivy/ui-components';
+import { useEffect, useRef, useState, type FocusEvent } from 'react';
+import { badgeProps } from '../../../utils/badge-properties';
+
+export type BadgeFieldProps = BasicFieldProps & {
+  value: string;
+  childRef: React.RefObject<HTMLElement>;
+  area?: boolean;
+};
+
+export const BadgeField = ({ value, label, children, message, childRef, area, ...props }: BadgeFieldProps) => {
+  const [focus, setFocus] = useState(false);
+  const inputRef = useRef<HTMLDivElement>(null);
+  const [height, setHeight] = useState(0);
+
+  const handleFocusLoss = (event: FocusEvent | MouseEvent) => {
+    console.log(event);
+    if (childRef.current && !childRef.current.contains(event.target as Node)) {
+      setFocus(false);
+    }
+  };
+
+  useEffect(() => {
+    setHeight(area ? splitNewLine(value).length * 14 : 15);
+    document.addEventListener('mousedown', handleFocusLoss);
+    return () => {
+      document.removeEventListener('mousedown', handleFocusLoss);
+    };
+  }, [value, area]);
+
+  const handleFocus = () => {
+    if (!focus) {
+      setFocus(true);
+      setTimeout(() => childRef.current!.focus(), 0);
+    }
+  };
```

Comment:
I would do this the same way we did it in the inscription view. with the useFocusWithin hook.